src/app/wallet/wallet/receive/receive.component.ts

```diff
@@ -359,4 +362,35 @@ export class ReceiveComponent implements OnInit {
     }
   }
 
+  updateLabel(address?: string) {
+    this.address = address ? address : '';
+    this.modals.unlock({timeout: 3}, (status) => address ? this.addNewLabel() : this.generateAddress());
+  }
+
+  generateAddress(): void {
+    const call = (this.type === 'public' ? 'getnewaddress' : (this.type === 'private' ? 'getnewstealthaddress' : ''));
+    const callParams = ['no label'];
+    const msg = `New ${this.type} address generated, with label no label!`;
+    this.callRpc(call, callParams, msg);
+  }
+
+  addNewLabel(): void {
+    const call = 'manageaddressbook';
+    const callParams = ['newsend', this.address, this.label];
+    const msg = `Updated label of ${this.address} to ${this.label}`;
+    this.callRpc(call, callParams, msg);
+  }
+
+  callRpc(call: string, callParams: any, msg: string): void {
+    if (call) {
+      this.rpc.call(call, callParams)
+        .subscribe(response => {
+          this.log.d(call, `addNewLabel: successfully executed ${call} ${callParams}`);
+          this.flashNotificationService.open(msg)
+          this.addressInput = true;
```

Comment:
I'm not really fond of the addressInput variable, it seems pretty useless.
Just directly manipulate the input field (by referencing it with #someId & viewchilds)